### PR TITLE
Add cmd flag to skip http paths from authentication

### PR DESCRIPTION
**Pull request by arajkumar**
Signed-off-by: Arunprasad Rajkumar <ar.arunprasad@gmail.com>

## Description

This PR adds a flag named `--web.auth.ignore-path` which takes a http path and passed path would be ignored from authentication. The flag shall be repeated to pass array of ignored paths.

e.g.
```
# skips authn for heathz and api/query_range endpoints.
./dist/promscale --web.auth.ignore-path=/heathz --web.auth.ignore-path=/api/query_range
```

Fixes https://github.com/timescale/promscale/issues/1636

Code changes are inspired from https://github.com/brancz/kube-rbac-proxy/blob/fc1ca4f969941340a8adb66932bd64dc5773d37a/main.go#L298-L321

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

File: pkg/api/router.go [hunk 1]
```diff
@@ -163,8 +164,22 @@ func authHandler(cfg *Config, handler http.Handler) http.Handler {
 		return handler
 	}
 
+	isIgnoredPath := func(r *http.Request) bool {
```
Comment (alejandrodnm) [inline pkg/api/router.go:167]: Maybe we can add this as a method to `Auth`?

**Review by alejandrodnm — APPROVED** [review]

**Review by antekresic — CHANGES_REQUESTED** [review]
Overall looks good, a bit of organization would benefit us here with the rise of complexity in the auth code.

**Comment by Harkishen-Singh** [comment]
I will pass this review to other reviewers.

**Review by alejandrodnm — APPROVED** [review]

File: pkg/api/router.go [hunk 2]
```diff
@@ -29,7 +28,7 @@ import (
 	"github.com/timescale/promscale/pkg/telemetry"
 )
 
-func GenerateRouter(apiConf *Config, promqlConf *query.Config, client *pgclient.Client, store *jaegerStore.Store, reload func() error) (*mux.Router, error) {
+func GenerateRouter(apiConf *Config, promqlConf *query.Config, client *pgclient.Client, store *jaegerStore.Store, authWrapper mux.MiddlewareFunc, reload func() error) (*mux.Router, error) {
```
Comment (antekresic) [inline pkg/api/router.go:31]: This is obviously a smell, having so many input parameters on a function. The function itself is too big as well.

Lets put a TODO on this to refactor it in the future so when somebody goes to add another parameter, they can see it and not do it 😃

**Review by antekresic — APPROVED** [review]
Approved with a small ask to add a TODO for some future refactoring.

**Pull request merged**